### PR TITLE
extend error message for chown function

### PR DESCRIPTION
**Pull request by Ruadhri17**
## Proposed changes
Small improvement to the error messages created by tedge init commands that was mentioned in #1795. Issue only happened when we wanted to change the owner/group of the file. The problem was with chown that handled errors by itself but did not provide any information about directory, file etc. 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#1795
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by didier-wenzek — COMMENTED** [review]
This is already a nice improvement. However, this can be improved a bit more.

Now, we have this error:
```
$ tedge --init        
Error: Failed to change owner: "/var/log/tedge".
```

But it would be good to the user what need to do using [`anyhow::with_context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context) in `tedge::initialize_tedge()` to get:

```
$ tedge --init
Error: Failed to initialize tedge. You have to run tedge with sudo.

Caused by:
      Failed to change owner: "/var/log/tedge".
```

**Review by PradeepKiruvale — COMMENTED** [review]
LGTM, I feel its good to add some unit tests.

**Comment by didier-wenzek** [comment]
> LGTM, I feel its good to add some unit tests.

One can add a system test, but I see no point to have unit tests.

**Comment by Ruadhri17** [comment]
> This is already a nice improvement. However, this can be improved a bit more.
> 
> Now, we have this error:
> 
> ```
> $ tedge --init        
> Error: Failed to change owner: "/var/log/tedge".
> ```
> 
> But it would be good to the user what need to do using [`anyhow::with_context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context) in `tedge::initialize_tedge()` to get:
> 
> ```
> $ tedge --init
> Error: Failed to initialize tedge. You have to run tedge with sudo.
> 
> Caused by:
>       Failed to change owner: "/var/log/tedge".
> ```

Changed for tedge and all plugins that have an initialization process.

File: plugins/c8y_configuration_plugin/src/main.rs [hunk 1]
```diff
@@ -73,7 +74,12 @@ async fn main() -> Result<(), anyhow::Error> {
     let tedge_config = config_repository.load()?;
 
     if config_plugin_opt.init {
-        init(config_plugin_opt.config_dir)
+        init(config_plugin_opt.config_dir).with_context(|| {
+            format!(
+                "Failed to initialize {}. You have to run command with sudo.",
```
Comment (didier-wenzek) [inline plugins/c8y_configuration_plugin/src/main.rs:79]:
```suggestion
                "Failed to initialize {}. You have to run the command with sudo.",
```

**Review by didier-wenzek — APPROVED** [review]
Approved.

Please fix the typo that is in all the error messages (missing the article *the*).

**Pull request merged**